openeo_driver/asset_urls.py

```diff
@@ -0,0 +1,37 @@
+import logging
+from openeo_driver.users import user_id_b64_encode
+import flask
+_log = logging.getLogger(__name__)
+
+
+class AssetUrl:
+    """
+    This class helps to convert internal storage locations to asset urls
+
+
+    If assets are not stored on local storage it is possible to come up with other implementations that extend this
+    class and go straight to the external storage. If the driver always has access to the artifacts then it is
+     recommended fallback to the default implementation by calling the method from its base class.
+    """
+    def build_url(self, asset_metadata: dict, asset_name: str, job_id: str, user_id: str) -> str:
```

Comment:
API design tip:
```suggestion
    def build_url(self, *, asset_metadata: dict, asset_name: str, job_id: str, user_id: str) -> str:
```

This forces all usage to use keyword arguments and prevents off-by-one positional arguments mistakes. 
Enforcing keyword arg usage also simplifies future API evolution